File: src/main/java/io/github/kuroka3/playermanager/Commands/BanID.java
package io.github.kuroka3.playermanager.Commands;

import io.github.kuroka3.playermanager.Utils.BanIDManager;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.block.CommandBlock;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.jetbrains.annotations.NotNull;
import org.json.simple.JSONObject;

import java.util.UUID;

public class BanID implements CommandExecutor {

    private static String mod = ChatColor.GRAY + "[" + ChatColor.LIGHT_PURPLE + "!" + ChatColor.GRAY + "] " + ChatColor.RESET;

    @Override
    public boolean onCommand(@NotNull CommandSender sd, @NotNull Command cmd, @NotNull String label, @NotNull String[] args) {
        try {
            if (args.length == 0) {
                sd.sendMessage(mod + ChatColor.RED + "BanID를 입력하십시오");
                return true;
            }

            if (args.length > 1) {
                sd.sendMessage(mod + ChatColor.RED + "불필요한 입력값을 제거하십시오");
                return true;
            }

            if (sd instanceof CommandBlock) {
                return true;
            }

            String BanID = args[0];

            JSONObject info = BanIDManager.getBanInfo(BanID);

            if (info == null) {
                sd.sendMessage(mod + ChatColor.RED + "해당 BanID를 찾을 수 없습니다");
                return true;
            }

            String moder = Bukkit.getOfflinePlayer((UUID.fromString((String) info.get("moder")))).getName();
            String target = Bukkit.getOfflinePlayer((UUID.fromString((String) info.get("target")))).getName();
            String reason = (String) info.get("reason");
            String date = (String) info.get("time");

            sd.sendMessage(ChatColor.GREEN + "=-=-=-=-=-=-=-=-=-=-=-=-=");
            sd.sendMessage(ChatColor.YELLOW + "BanID" + ChatColor.GRAY + " :: " + ChatColor.GREEN + BanID);
            sd.sendMessage(ChatColor.YELLOW + "Moder" + ChatColor.GRAY + " :: " + ChatColor.GREEN + moder);
            sd.sendMessage(ChatColor.YELLOW + "Target" + ChatColor.GRAY + " :: " + ChatColor.GREEN + target);
            sd.sendMessage(ChatColor.YELLOW + "Reason" + ChatColor.GRAY + " :: " + ChatColor.GREEN + reason);
            sd.sendMessage(ChatColor.YELLOW + "Time" + ChatColor.GRAY + " :: " + ChatColor.GREEN + date);
            sd.sendMessage(ChatColor.GREEN + "=-=-=-=-=-=-=-=-=-=-=-=-=");

            return true;
        } catch (Exception e) {
            e.printStackTrace();
            return true;
        }
    }
}

File: src/main/kotlin/io/github/kuroka3/playermanagerkotlin/Commands/BanID.kt
package io.github.kuroka3.playermanagerkotlin.Commands

import io.github.kuroka3.playermanagerkotlin.PlayerManagerKotlin
import io.github.kuroka3.playermanagerkotlin.Utils.BanIDManager
import io.github.monun.kommand.getValue
import io.github.monun.kommand.kommand
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.Component.text
import net.kyori.adventure.text.TextComponent
import net.kyori.adventure.text.format.TextColor
import org.bukkit.Bukkit
import org.json.simple.JSONObject
import java.util.*

object BanID {
    fun registerKommand() {

        val PlayerManagerKotlin = PlayerManagerKotlin.instance

        val mod: TextComponent = text("[").color(TextColor.color(0x00aaaaaa)).append(
            text("!").color(TextColor.color(0x00ff55ff))
        ).append(
            text("] ").color(TextColor.color(0x00aaaaaa))
        )

        PlayerManagerKotlin.kommand {
            register("banid") {
                then("id" to string()) {
                    executes {
                        val id: String by it
                        val info: JSONObject? = BanIDManager.getBanInfo(id)

                        if(info == null) {
                            sender.sendMessage(mod.append(text("해당 BanID를 찾을 수 없습니다").color(TextColor.color(0x00ff55555))))
                            return@executes
                        }

                        val moder = if(info["moder"] == "Server") "Server" else Bukkit.getOfflinePlayer(UUID.fromString(info["moder"].toString())).name!!
                        val target = Bukkit.getOfflinePlayer(UUID.fromString(info["target"].toString())).name!!
                        val reason = info["reason"] as String
                        val date = info["time"] as String
                        val isTemp = info["isTemp"] as Boolean
                        val period = info["period"] as String?

                        sender.sendMessage(text("\n=-=-=-=-=-=-=-=-=-=-=-=-=").color(TextColor.color(0x0055ff55)))
                        sender.sendMessage(text("BanID").color(TextColor.color(0x00ffff55)).append(
                            text(" :: ").color(TextColor.color(0x00aaaaaa))).append(
                            text(id).color(TextColor.color(0x0055ff55))))
                        sender.sendMessage(text("Moder").color(TextColor.color(0x00ffff55)).append(
                            text(" :: ").color(TextColor.color(0x00aaaaaa))).append(
                            text(moder).color(TextColor.color(0x0055ff55))))
                        sender.sendMessage(text("Target").color(TextColor.color(0x00ffff55)).append(
                            text(" :: ").color(TextColor.color(0x00aaaaaa))).append(
                            text(target).color(TextColor.color(0x0055ff55))))
                        sender.sendMessage(text("Reason").color(TextColor.color(0x00ffff55)).append(
                            text(" :: ").color(TextColor.color(0x00aaaaaa))).append(
                            text(reason).color(TextColor.color(0x0055ff55))))
                        sender.sendMessage(text("Time").color(TextColor.color(0x00ffff55)).append(
                            text(" :: ").color(TextColor.color(0x00aaaaaa))).append(
                            text(date).color(TextColor.color(0x0055ff55))))
                        sender.sendMessage(text("isTempBan").color(TextColor.color(0x00ffff55)).append(
                            text(" :: ").color(TextColor.color(0x00aaaaaa))).append(
                            text(isTemp).color(TextColor.color(0x0055ff55))))
                        if(period != null) sender.sendMessage(text("Period").color(TextColor.color(0x00ffff55)).append(
                            text(" :: ").color(TextColor.color(0x00aaaaaa))).append(
                            text(period).color(TextColor.color(0x0055ff55))))
                        sender.sendMessage(text("=-=-=-=-=-=-=-=-=-=-=-=-=\n").color(TextColor.color(0x0055ff55)))
                    }

                    then("asUUID") {
                        executes {
                            val id: String by it
                            val info: JSONObject? = BanIDManager.getBanInfo(id)

                            if(info == null) {
                                sender.sendMessage(mod.append(text("해당 BanID를 찾을 수 없습니다").color(TextColor.color(0x00ff55555))))
                                return@executes
                            }

                            val moder = info["moder"] as String
                            val target = info["target"] as String
                            val reason = info["reason"] as String
                            val date = info["time"] as String
                            val isTemp = info["isTemp"] as Boolean
                            val period = info["period"] as String?

                            sender.sendMessage(text("\n=-=-=-=-=-=-=-=-=-=-=-=-=").color(TextColor.color(0x0055ff55)))
                            sender.sendMessage(text("BanID").color(TextColor.color(0x00ffff55)).append(
                                text(" :: ").color(TextColor.color(0x00aaaaaa))).append(
                                text(id).color(TextColor.color(0x0055ff55))))
                            sender.sendMessage(text("Moder").color(TextColor.color(0x00ffff55)).append(
                                text(" :: ").color(TextColor.color(0x00aaaaaa))).append(
                                text(moder).color(TextColor.color(0x0055ff55))))
                            sender.sendMessage(text("Target").color(TextColor.color(0x00ffff55)).append(
                                text(" :: ").color(TextColor.color(0x00aaaaaa))).append(
                                text(target).color(TextColor.color(0x0055ff55))))
                            sender.sendMessage(text("Reason").color(TextColor.color(0x00ffff55)).append(
                                text(" :: ").color(TextColor.color(0x00aaaaaa))).append(
                                text(reason).color(TextColor.color(0x0055ff55))))
                            sender.sendMessage(text("Time").color(TextColor.color(0x00ffff55)).append(
                                text(" :: ").color(TextColor.color(0x00aaaaaa))).append(
                                text(date).color(TextColor.color(0x0055ff55))))
                            sender.sendMessage(text("isTempBan").color(TextColor.color(0x00ffff55)).append(
                                text(" :: ").color(TextColor.color(0x00aaaaaa))).append(
                                text(isTemp).color(TextColor.color(0x0055ff55))))
                            if(period != null) sender.sendMessage(text("Period").color(TextColor.color(0x00ffff55)).append(
                                text(" :: ").color(TextColor.color(0x00aaaaaa))).append(
                                text(period).color(TextColor.color(0x0055ff55))))
                            sender.sendMessage(text("=-=-=-=-=-=-=-=-=-=-=-=-=\n").color(TextColor.color(0x0055ff55)))
                        }
                    }
                }
            }
        }
    }
}